enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -66,4 +77,75 @@ public static EndpointCluster processEndpoints(org.wso2.choreo.connect.discovery
         }
         return endpointCluster;
     }
+
+    /**
+     * Handles mock API call and prepares response object considering provided values in the request.
+     *
+     * @param requestContext request context
+     * @param responseObject response object for the mock API call
+     */
+    public static void processMockedApiCall(RequestContext requestContext, ResponseObject responseObject) {
+        responseObject.setDirectResponse(true);
+        String acceptType = "";
+        MockedApiConfig mockedApiConfig = requestContext.getMatchedResourcePath().getMockedApiConfig();
+        Map<String, String> headersMap = requestContext.getHeaders();
+        if (headersMap.containsKey(APIConstants.ACCEPT_HEADER.toLowerCase())) {
+            acceptType = headersMap.get(APIConstants.ACCEPT_HEADER.toLowerCase());
+        }
+
+        if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            setMockApiResponse(responseObject, headersMap, mockedApiConfig, acceptType, false);
+        } else if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.QUERY)) {
+            Map<String, String> queryParamMap = requestContext.getQueryParameters();
+            setMockApiResponse(responseObject, queryParamMap, mockedApiConfig, acceptType, false);
+        } else {
+            setMockApiResponse(responseObject, null, mockedApiConfig, acceptType, true);
+        }
+    }
+
+    /**
+     * Prepares mock API response considering the properties map and mockedApiConfig.
+     *
+     * @param responseObject  Response object (represents response for the API call)
+     * @param propertiesMap   A map which includes values specified in headers or query parameters
+     * @param mockedApiConfig Holds the JSON values specified in the mock API implementation
+     * @param acceptType      Denotes accepted contend type as the response
+     */
+    private static void setMockApiResponse(ResponseObject responseObject, Map<String, String> propertiesMap,
+                                           MockedApiConfig mockedApiConfig, String acceptType, boolean isDefault) {
+        String requestValuePosition = mockedApiConfig.getIn();
+        String nameFieldInJson;
+
+        // condition handles case-sensitiveness of query parameters.
+        if (requestValuePosition.equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            nameFieldInJson = mockedApiConfig.getName().toLowerCase();
+        } else {
+            nameFieldInJson = mockedApiConfig.getName();
+        }
+
+        if (!isDefault && (propertiesMap == null || !propertiesMap.containsKey(nameFieldInJson))) {
+            log.error("Response determining value not available in the mock API request.");
```

Comment:
Whats going to be the response to the client in this scenario?

Reply:
Previously there was an error log only. But with https://github.com/wso2/product-microgateway/pull/2620/commits/cb025dbd99e481a384431940eecce3336ff0d6a2 added an error log to display as the response message.